### PR TITLE
feat: include sbom yaml to build SBOM

### PR DESCRIPTION
**Pull request by nathansandi**
## Description

Include SBOM action for generate artefacts independent of pom.xml

- Remove Cyclodex of locally build process
- Re-include the Github workflow

## Checklist

- [ ] for CI changes:
  - [ ] structural/foundational changes signed off by [CI DRI](https://github.com/cmur2)
  - [ ] [ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml) modifications comply with ["Unified CI" requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)

## Related issues

closes #

### CODE REVIEW

File: .github/workflows/sbom.yaml [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+# This workflow generates a Software Bill of Materials (SBOM) using the CycloneDX Maven plugin.
+# The SBOM provides a list of dependencies and helps track potential security risks.
+# The workflow runs on a schedule for testing purposes and can be manually triggered.
+# The generated SBOM is uploaded as an artifact for further analysis or distribution.
+
+name: Generate SBOM
+
+on:
+  workflow_dispatch:
+  pull_request: # Triggers the workflow on pull requests
```
Comment (vsgoulart) [inline .github/workflows/sbom.yaml:11]: Why are we triggering in the pull request again? We had removed it in the final PR https://github.com/camunda/camunda/pull/23070/files

Comment (vsgoulart) [reply .github/workflows/sbom.yaml:11]: Ah, I guess you added it again to test?

Comment (nathansandi) [reply .github/workflows/sbom.yaml:11]: Yes :) Will remove it before the review

**Comment by clementnero** [comment]
@nathansandi @vsgoulart 
To prevent [INC-1845](https://app.incident.io/camunda/incidents/1845) from happening again and to ensure that the PR is free of compilation errors on Infra runners, I would suggest disabling the maven cache with a test [commit](https://github.com/camunda/camunda/commit/d3ea352e47845517c10b1f47f663b3255882f2bf) (can be cherry-picked temporarily) and checking that all jobs pass (in particular for `ci.yml` and `zeebe-ci.yml` workflows). This will probably be 
I have very limited availability today due to team offsite, but can take some time tomorrow if needed.

**Comment by clementnero** [comment]
> On my understanding one of the points that increased the build time on the CI was the fact was introduced the cyclodex over the pom dependencies and it was being executed over the mvn install job, which was incresing the build time of the jobs and causing timeout. But this was a problem apart.

Yes. The build time was one of the two issues. The second issue was very weird compilation errors only happening on Infra runners and if no cache exists. Probably an edge-case/bug caused by the plugin.

Now that cyclodex is not introduced in the pom dependencies and Infra runners is not used to execute the `backend_sbom` job, it safe to say that this PR will not reintroduce a similar issue as [INC-1845](https://app.incident.io/camunda/incidents/1845).

**Review by clementnero — APPROVED** [review]
Regarding [INC-1845](https://app.incident.io/camunda/incidents/1845), it looks good to me and safe to merge.

Thank you!

**Comment by clementnero** [comment]
@nathansandi Should this PR be backported to stable/8.6? Just want to check this with you before closing the incident.

**Pull request merged**